### PR TITLE
websocket: deprecating old style websocket

### PR DESCRIPTION
**Pull request by alyssawilk**
I think timeouts were the last critical feature - please correct me if I'm wrong.

I'm also happy to wait until folks have moved over, but note that deprecation simply means we'll clean up the old path in 1.5-4.5 months in stead of 4.5-7.5

*Risk Level*: n/a
*Testing*: added timeout test for websocket
*Docs Changes*: added deprecation notice
*Release Notes*: n/a

### CODE REVIEW

**Comment by ggreenway** [comment]
Shouldn't the config in the .proto's get tagged with [deprecated = true] ?

File: test/integration/websocket_integration_test.cc [hunk 1]
```diff
@@ -237,27 +237,20 @@ TEST_P(WebsocketIntegrationTest, EarlyData) {
...
-          auto* route_idle_timeout = route->mutable_route()->mutable_timeout();
-          auto nanos =
-              std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::milliseconds(100))
-                  .count();
-          route_idle_timeout->set_nanos(nanos);
-          auto* idle_timeout = hcm.mutable_idle_timeout();
-          idle_timeout->set_nanos(nanos);
+        [&](envoy::config::filter::network::http_connection_manager::v2::HttpConnectionManager& hcm)
+            -> void {
+          auto* route_config = hcm.mutable_route_config();
```
Comment (ggreenway) [inline test/integration/websocket_integration_test.cc]: This change seems unrelated to deprecating this. Should it be in a separate PR?

Comment (alyssawilk) [reply test/integration/websocket_integration_test.cc]: If you'd prefer I'd be happy to.  The thing holding me back from deprecation was the lack of idle timeouts, so to me it's finishing the feature and deprecating the old version.  If we think it's worth splitting out it's easy enough on my end.

**Comment by rshriram** [comment]
Is websocket arch doc updated ?

**Comment by alyssawilk** [comment]
I annotated new and old style here:
https://github.com/envoyproxy/envoy/pull/3776/files

I could explicitly call out old style as deprecated as part of this PR if you think that's worth it?
